src/ir_Teco.cpp

```diff
@@ -0,0 +1,264 @@
+  newtemp -= kTecoMinTemp;  // 16=0b000
+
+  remote_state &= ~kTecoTempMask;  // reinit temp
+  remote_state |= (newtemp << 8);
+}
+
+uint8_t IRTecoAC::getTemp() {
+  return ((remote_state & kTecoTempMask) >> 8) + kTecoMinTemp;
+}
+
+// Set the speed of the fan
+void IRTecoAC::setFan(uint8_t speed) {
+  uint8_t newspeed = speed;
+  switch (speed) {
+    case kTecoFanAuto:
+    case kTecoFan3:
+    case kTecoFan2:
+    case kTecoFan1:
+      break;
+    default:
+      newspeed = kTecoFanAuto;
+  }
+  remote_state &= ~kTecoFanMask;  // reinit fan
+  remote_state |= (newspeed << 4);
+}
+
+uint8_t IRTecoAC::getFan() {
+  return (remote_state & kTecoFanMask) >> 4;
+}
+
+void IRTecoAC::setMode(uint8_t mode) {
+  uint8_t newmode = mode;
+  switch (mode) {
+    case kTecoAuto:
+    case kTecoCool:
+    case kTecoDry:
+    case kTecoFan:
+    case kTecoHeat:
+      break;
+    default:
+      newmode = kTecoAuto;
+  }
+  remote_state &= ~kTecoModeMask;  // reinit mode
+  remote_state |= newmode;
+}
+
+uint8_t IRTecoAC::getMode() {
+  return remote_state & kTecoModeMask;
+}
+
+void IRTecoAC::setSwing(bool state) {
+  if (state) remote_state |= kTecoSwing;
+  else
+    remote_state &= ~kTecoSwing;
+}
+
+bool IRTecoAC::getSwing() {
+  return (remote_state & kTecoSwing) == kTecoSwing;
+}
+
+void IRTecoAC::setSleep(bool state) {
+  if (state) remote_state |= kTecoSleep;
+  else
+    remote_state &= ~kTecoSleep;
+}
+
+bool IRTecoAC::getSleep() {
+  return (remote_state & kTecoSleep) == kTecoSleep;
+}
+
+// Convert the internal state into a human readable string.
+#ifdef ARDUINO
+String IRTecoAC::toString() {
+  String result = "";
+#else
+std::string IRTecoAC::toString() {
+  std::string result = "";
+#endif  // ARDUINO
+  result += "Power: ";
+  if (getPower()) {
+    result += "On";
+  } else {
+    result += "Off";
+    return result;  // If it's off, there is no other info.
+  }
+  result += ", Fan: " + uint64ToString(getFan());
+  switch (getFan()) {
+    case kTecoFanAuto:
+      result += " (AUTO)";
+      break;
+    case kTecoFan3:
+      result += " (MAX)";
+      break;
+    case kTecoFan1:
+      result += " (MIN)";
+      break;
+    case kTecoFan2:
+      result += " (MED)";
+      break;
+    default:
+      result += " (UNKNOWN)";
+  }
+  // Special modes.
+  if (getSwing()) {
+    result += ", Swing: Toggle";
+    return result;
+  }
+  if (getSleep()) {
+    result += ", Sleep: Toggle";
+    return result;
+  }
+  result += ", Mode: " + uint64ToString(getMode());
+  switch (getMode()) {
+    case kTecoAuto:
+      result += " (AUTO)";
+      break;
+    case kTecoCool:
+      result += " (COOL)";
+      break;
+    case kTecoHeat:
+      result += " (HEAT)";
+      break;
+    case kTecoDry:
+      result += " (DRY)";
+      break;
+    case kTecoFan:
+      result += " (FAN)";
+      break;
+    default:
+      result += " (UNKNOWN)";
+  }
+  if (getMode() != kTecoFan)  // Fan mode doesn't have a temperature.
```

Comment:
I'm guessing the remote doesn't display the temp when in `kTecoFan` mode, but it might be wise to show this value, unless it is some special value when in fan mode. Is there a special value for the temp when in Fan mode? (see earlier comment)

Reply:
No, Fan mode does not change the temperature bits